### DISCUSSION TITLE
How can I scroll the page in copy mode within another app?

### DISCUSSION THREAD

**Discussion by azzamsa** [wezterm/wezterm#4763] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Gnome

### WezTerm version

20230712-072601-f4abf8fd

### Ask your question!

How to scroll in `copy_mode` inside another app with scroll capabilities? Especially [Bacon](https://dystroy.org/bacon/).

When I activate `copy_mode` the code below doesn't work.

```lua
keys = {
    { key = "y", mods = "ALT", action = "ActivateCopyMode" },
    { key = "PageUp", mods = "SHIFT", action = act.ScrollByPage(-1) },
    { key = "PageDown", mods = "SHIFT", action = act.ScrollByPage(1) },
     ... other code
```

```lua
copy_mode = {
     ... other code
    { key = "PageUp", action = act.ScrollByPage(-0.5) },
    { key = "PageDown", action = act.ScrollByPage(0.5) },
},
```

![image](https://github.com/wez/wezterm/assets/17734314/9dd8b002-dded-49af-a71e-3173e933cfc9)

**Comment by wez** [comment, marked as answer]
"Full screen" TUI programs typically enable the alt-screen which has no scrollback, so there is nothing to scroll.
You may be able to configure the program to not use the alt-screen; that is something you need to figure out for each application.

**Reply by azzamsa** [reply]
I will keep using `cargo check` to share the output with others.
